src/main/java/org/beanpod/switchboard/config/DataSourceConfig.java

```diff
@@ -0,0 +1,56 @@
+package org.beanpod.switchboard.config;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import javax.sql.DataSource;
+import org.springframework.boot.jdbc.DataSourceBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+
+@Configuration
+public class DataSourceConfig {
+
+  @Bean(name = "h2DataSource")
+  @Profile("test")
+  public DataSource h2DataSource() {
+    DataSourceBuilder<?> dataSourceBuilder = DataSourceBuilder.create();
+    dataSourceBuilder.driverClassName("org.h2.Driver");
+    dataSourceBuilder.url("jdbc:h2:mem:test");
+    dataSourceBuilder.username("sa");
+    dataSourceBuilder.password("");
```

Comment:
What are these credentials from? Are they even used?

Reply:
These are the default h2 database credentials. It's common practice to use the defaults for a testing db.